camel/models/base_model.py

```diff
@@ -44,11 +45,13 @@ class BaseModelBackend(ABC):
     def __init__(
         self,
         model_type: ModelType,
-        model_config_dict: Dict[str, Any],
+        model_config_dict: Optional[Dict[str, Any]] = None,
         api_key: Optional[str] = None,
         url: Optional[str] = None,
         token_counter: Optional[BaseTokenCounter] = None,
     ) -> None:
+        if not model_config_dict:
+            model_config_dict = {}
```

Comment:
I think the model_config_dict can be not `Optional` (and also not optional w/o default values) in `BaseModel` because it will be constructed from subclasses, which should be responsible for setting the value of it.